compiler.rst

```diff
@@ -159,49 +159,81 @@ are very rare.  However, if you've allocated a PyObject, you must tell
 the arena about it by calling ``PyArena_AddPyObject()``.
 
 
-Parse Tree to AST
------------------
-
-The AST is generated from the parse tree (see :file:`Python/ast.c`) using the
-function ``PyAST_FromNode()``.
+Tokens to AST
```

Comment:
Using tokens here is a bit confusing because the source of all the "public" APIs used by interpreter takes the source code as a string buffer, and in the process of parsing, it transforms it to tokens. I suggest to change these occurrences to "source code to AST".

Reply:
Thank you. I was initially confused too but I was hesitant to change it -- lots of the online diagrams I read didn't seem to actually reflect what the source code was doing. Eventually I just gave up and stepped through the source code to write these docs :P.